sqlitedict.py

```diff
@@ -356,6 +362,31 @@ def __del__(self):
             # in __del__ method.
             pass
 
+def _convertkey(key):
+    if isinstance(key,int):
+        return '___.int.' + repr(key)
+    elif isinstance(key,float):
+        return '___.float.' + repr(key)
+    # These are recursive
+    elif isinstance(key,tuple):
+        return '___.tuple.' + json.dumps([_convertkey(k) for k in key])
```

Comment:
It would be very helpful to allow passing in a JSONEncoder to be used here as a `cls=..` argument, to allow inclusion of types which need help serialising.

This could be done after your PR, if you like, but that may be a useful element in the design of this PR.